hnn_core/extracellular.py

```diff
@@ -28,6 +28,75 @@
 from .externals.mne import _validate_type, _check_option
 
 
+def _calculate_csd2d(lfp_data, ch_axis=0, delta=1):
+    """Current source density (CSD) estimation
+
+    The three-point finite-difference approximation of the
+    second spatial derivative for computing 1-dimensional CSD
+    with border electrode interpolation
+    csd[electrode] = -(LFP[electrode - 1] - 2*LFP[electrode] +
+                       LFP[electrode + 1]) / spacing ** 2
+
+    Parameters
+    ----------
+    lfp_data : channels x times array
+        LFP data.
+    ch_axis : int
+        Axis of electrode array, default is zero.
+    delta : int
+        Spacing between channels, scales the CSD.
+
+    Returns
+    -------
+    csd2d : channels x times array
+        the 2nd derivative current source density estimate (csd2d)
+    """
+    csd2d = -np.diff(np.diff(lfp_data, axis=0), axis=0) / delta ** 2
+    bottom_border = csd2d[-1, :] * 2 - csd2d[-2, :]
+    top_border = csd2d[0, :] * 2 - csd2d[1, :]
+    csd2d = np.concatenate([np.expand_dims(top_border, axis=ch_axis),
```

Comment:
not sure what `expand_dims` does? I normally just do `x[None, ...]` if I want to add another dimension. It also is a bit more clear IMO.

Reply:
I'm not actually sure what `np.expand_dims()` does. Maybe @spbrandt can weigh in here?

Is `x[None, ...]` the same as `x[np.newaxis, ...]`? If so, is one better than the other?